File: ship-ui/src/modal/entiy/Department.ts
import {Page} from "@/modal/DO/Page";

export interface Department {
    departmentId: number;
    departmentName: string;
    departmentInfo: string;
    departmentAddress: string;
    createTime?: null | string;
    updateTime?: null | string;
    isDel?: null | string;
}

/**
 * 后台列表的部门
 */
export const exampleDepartBack = (): Page<Department> => {
    return {
        "records": [
            {
                "departmentId": 1,
                "departmentName": "犬猫科",
                "departmentInfo": "主要整治猫猫狗狗",
                "departmentAddress": "一楼227",
                "createTime": null,
                "updateTime": null,
                "isDel": null
            },
            {
                "departmentId": 2,
                "departmentName": "骨科",
                "departmentInfo": "专治跌打损伤",
                "departmentAddress": "三楼238",
                "createTime": null,
                "updateTime": null,
                "isDel": null
            },
            {
                "departmentId": 3,
                "departmentName": "化验室",
                "departmentInfo": "进行化验",
                "departmentAddress": "三楼226",
                "createTime": null,
                "updateTime": null,
                "isDel": null
            },
            {
                "departmentId": 4,
                "departmentName": "美容室",
                "departmentInfo": "对宠物进行美容",
                "departmentAddress": "三楼327",
                "createTime": null,
                "updateTime": null,
                "isDel": null
            },
            {
                "departmentId": 8,
                "departmentName": "中兽科",
                "departmentInfo": "传统中医治兽",
                "departmentAddress": "一楼125",
                "createTime": null,
                "updateTime": null,
                "isDel": null
            }
        ],
        "total": 5
    }
}

File: shipPet/src/main/java/com/ship/model/entity/Department.java
package com.ship.model.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.io.Serial;

/**
 * <p>
 * 部门表
 * </p>
 *
 * @author MixJade
 * @since 2023-08-22
 */
public class Department implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 科室id
     */
    @TableId(value = "department_id", type = IdType.AUTO)
    private Integer departmentId;

    /**
     * 科室名称
     */
    private String departmentName;

    /**
     * 科室简介
     */
    private String departmentInfo;

    /**
     * 科室地址
     */
    private String departmentAddress;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 修改时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 逻辑删除，默认0，填充删除日期
     */
    private Long isDel;

    public Integer getDepartmentId() {
        return departmentId;
    }

    public void setDepartmentId(Integer departmentId) {
        this.departmentId = departmentId;
    }

    public String getDepartmentName() {
        return departmentName;
    }

    public void setDepartmentName(String departmentName) {
        this.departmentName = departmentName;
    }

    public String getDepartmentInfo() {
        return departmentInfo;
    }

    public void setDepartmentInfo(String departmentInfo) {
        this.departmentInfo = departmentInfo;
    }

    public String getDepartmentAddress() {
        return departmentAddress;
    }

    public void setDepartmentAddress(String departmentAddress) {
        this.departmentAddress = departmentAddress;
    }

    public LocalDateTime getCreateTime() {
        return createTime;
    }

    public void setCreateTime(LocalDateTime createTime) {
        this.createTime = createTime;
    }

    public LocalDateTime getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(LocalDateTime updateTime) {
        this.updateTime = updateTime;
    }

    public Long getIsDel() {
        return isDel;
    }

    public void setIsDel(Long isDel) {
        this.isDel = isDel;
    }

    @Override
    public String toString() {
        return "Department{" +
                "departmentId = " + departmentId +
                ", departmentName = " + departmentName +
                ", departmentInfo = " + departmentInfo +
                ", departmentAddress = " + departmentAddress +
                ", createTime = " + createTime +
                ", updateTime = " + updateTime +
                ", isDel = " + isDel +
                "}";
    }
}
